source/PlanetPanel.cpp

```diff
@@ -246,20 +248,58 @@ void PlanetPanel::TakeOffIfReady()
 	}
 
 	// Check if any of the player's ships are configured in such a way that they
-	// will be impossible to fly.
-	const auto flightChecks = player.FlightCheck();
+	// will be impossible to fly. If so, let the player choose whether to park them.
+	ostringstream out;
+	flightChecks = player.FlightCheck();
 	if(!flightChecks.empty())
+	{
 		for(const auto &result : flightChecks)
 		{
-			// If there is a flightcheck error, it will be the first (and only) entry.
+			// If there is a flight check error, it will be the first (and only) entry.
 			auto &check = result.second.front();
 			if(check.back() == '!')
 			{
-				GetUI()->Push(new ConversationPanel(player,
-					*GameData::Conversations().Get("flight check: " + check), nullptr, result.first));
-				return;
+				// If the ship with a flight check error is in another system, then the only thing the player
+				// can do is park it. But if the ship is with the player, then they may be able to make changes
+				// to rectify the error. As such, provide a conversation for any single present ship, but
+				// record and report all absent ships later.
+				if(result.first->GetSystem() != &system)
+				{
+					out << result.first->Name() << ", ";
+					absentCannotFly.push_back(result.first);
+				}
+				else
+				{
+					GetUI()->Push(new ConversationPanel(player,
+						*GameData::Conversations().Get("flight check: " + check), nullptr, result.first));
+					return;
+				}
 			}
 		}
+		if(!absentCannotFly.empty())
+		{
+			string shipNames = out.str();
+			// Pop back the last ", " in the string.
+			shipNames.pop_back();
+			shipNames.pop_back();
+			GetUI()->Push(new Dialog(this, &PlanetPanel::CheckWarningsAndTakeOff,
+				"Some of your ships that are not with you are not be able to fly:\n" + shipNames +
```

Comment:
```suggestion
				"Some of your ships in other systems are not be able to fly:\n" + shipNames +
```
I feel like its clearer like that